src/ChatClient.ts

```diff
@@ -62,6 +63,7 @@ export class ChatClient {
         this.client.on("disconnect", () => { this.fireStatusChange(ConnectionStatus.DISCONNECTED) });
         this.client.on("error", (error) => {
             console.log(`Error: ${error}`);
+            sampleRUM('chat:client', { source: 'error', target: error });
```

Comment:
Don't duplicate information. I'd go with `chat:error` source would be the script or function, target the error message